Helper/IntegrationManagement.php

```diff
@@ -0,0 +1,468 @@
+                    IntegrationModel::SETUP_TYPE => IntegrationModel::TYPE_CONFIG,
+                ];
+                $boltIntegration = $this->integrationService->findByName(self::BOLT_INTEGRATION_NAME);
+                if (!$boltIntegration->getId()) {
+                    $boltIntegration = $this->integrationService->create($integrationData);
+                } else {
+                    $integrationData[IntegrationModel::ID] = $boltIntegration->getId();
+                    $boltIntegration = $this->integrationService->update($integrationData);
+                }
+            }  
+        } catch (Exception $e) {
+            $this->bugsnag->notifyException($e);
+        } finally {
+            return $boltIntegration;
+        }
+    }
+    
+    /**
+     * Link access token of Magento integration to Bolt merchant account via OAuth handshake.
+     *
+     * @param int|string $storeId
+     *
+     * @return bool
+     */
+    public function linkAccessTokenBoltMerchantAccountByOAuth($storeId = null)
+    {
+        $boltIntegration = $this->processMagentoIntegraion();
+        if (!$boltIntegration || !$boltIntegration->getId()) {
+            return false;
+        }
+        try {
+            if ($this->getIntegrationStatus($storeId) == self::BOLT_INTEGRATION_STATUS_CREATED) {
+                // Re-send API keys to Bolt
+                // For security reason, we have to remove existing token associated with consumer and issue a new one
+                $this->oauthService->deleteIntegrationToken($boltIntegration->getConsumerId());
+                $boltIntegration->setStatus(IntegrationModel::STATUS_INACTIVE)->save();
+            }
+            // Integration chooses to use Oauth for token exchange
+            // Execute post to integration (consumer) HTTP Post URL (endpoint_url).
+            $this->oauthService->postToConsumer($boltIntegration->getConsumerId(), $boltIntegration->getEndpoint());
+            // Ignore calling the page defined in the Identity Link,
+            // and send oauth_consumer_key and callback_url directly for token exchange.
+            $consumer = $this->oauthService->loadConsumer($boltIntegration->getConsumerId());
+            if (!$consumer->getId()) {
+                throw new Exception(
+                    __(
+                        'A consumer with "%1" ID doesn\'t exist. Verify the ID and try again.',
+                        $boltIntegration->getConsumerId()
+                    )
+                );
+            }
+            $loginSuccessCallback = $this->urlBuilder->getUrl('*/*/loginSuccessCallback');
+            $this->httpClient->setUri($this->getTokensExchangeUrl($storeId));
+            $this->httpClient->setParameterPost(
+                [
+                    'oauth_consumer_key' => $consumer->getKey(),
+                    'callback_url' => $loginSuccessCallback,
+                ]
+            );
```

Comment:
We should just send the data directly since merchant is authorizing the oauth by clicking on the button.
```suggestion
            $verifier = $this->tokenFactory->create()->createVerifierToken($boltIntegration->getConsumerId());
            $this->httpClient->setUri($this->getTokensExchangeUrl($storeId));
            $this->httpClient->setParameterGet(
                [
                    'oauth_consumer_secret' => $consumer->getSecret(),
                    'oauth_consumer_key' => $consumer->getKey(),
                    'store_base_url' => $this->urlBuilder->getUrl(),
                    'oauth_verifier' => $verifier->getVerifier(),
                ]
            );
```